### PR TITLE
Rainfall: Update navigation styles and tidy up templates

### LINKED ISSUE #6245: Rainfall: Header and Footer improvements
Here are some comments related with the header and footer.

1. Both the header and footer template parts should be full width, aligning with the post featured images.

Captura de ecrã 2022-07-15, às 11 06 12
Captura de ecrã 2022-07-15, às 11 08 49

2. The footer text should be uppercase by default, like the header text.

Captura de ecrã 2022-07-15, às 11 10 59

3. You probably haven't yet addressed the mobile menu but wanted to mention it anyway. Is there any issue open in Gutenberg about the customisation of the mobile menu icon? It would be neat to have this be customised "Menu" and "Close" instead of the default hamburger and `X` icons.

Captura de ecrã 2022-07-15, às 11 13 04

4. Lastly, there's some weirdness in the footer on mobile.5. 

Captura de ecrã 2022-07-15, às 11 19 41

### LINKED ISSUE #6431: Rainfall: "Menu" term should be the same size has site title
Currently:

Captura de ecrã 2022-08-23, às 10 46 58

Intended:

Captura de ecrã 2022-08-23, às 10 47 47

### PR DESCRIPTION
**Pull request by mikachan**
#### Changes proposed in this Pull Request:
This PR fixes the navigation styles on Rainfall to match the [designs](oz0a2czNn0UNt9onrh8T5a-fi-107%3A906) more closely. 

It also includes the following updates:

- adds the theme description from [here](https://github.com/Automattic/themes/issues/6421#issuecomment-1222095777)
- moves some CSS into theme.json
- adjusts the fluid typography settings to include some minimum values
- applies spacing presets across templates where possible
- updates all layout settings to new format (part of https://github.com/Automattic/themes/issues/6477)

#### Related issue(s):
https://github.com/Automattic/themes/pull/6305

### CODE REVIEW

File: rainfall/style.css [hunk 1]
```diff
@@ -1,25 +1,38 @@
...
 Text Domain: rainfall
-Tags: one-column, custom-colors, custom-menu, custom-logo, editor-style, featured-images, full-site-editing, rtl-language-support, theme-options, threaded-comments, translation-ready, wide-blocks
+Tags: two-columns, custom-colors, custom-menu, custom-logo, editor-style, featured-images, full-site-editing, block-patterns, rtl-language-support, theme-options, threaded-comments, translation-ready, auto-loading-homepage, blog-homepage
 */
 
+/*
+ * Font smoothing
+ */
+body {
+	-moz-osx-font-smoothing: grayscale;
```
Comment (scruffian) [inline rainfall/style.css]: I don't think we should be doing this.

**Comment by scruffian** [comment]
With this change my header now has additional padding:

Screenshot 2022-09-05 at 10 38 15

**Comment by mikachan** [comment]
I've just pushed a change that tweaks the fluid font sizes a bit more, I thought the range of sizes was too big as the small font size was looking quite big at larger resolutions.

> With this change my header now has additional padding:

I think this is because I've added left and right root padding values so we can use `useRootPaddingAwareAlignments`, and it looks like this is adding horizontal padding to the header. Mine doesn't look quite as bad/spaced out as yours, although I am seeing more padding on the align wide heading:

| Without this PR | With this PR |
| --------------- | ------------- |
| image | image | 
| image | image |

File: rainfall/style.css (hunk 1 above)
Comment (mikachan) [reply rainfall/style.css]: Yeah I'm not sure about this either! I feel like it's only there to improve font appearance for a small number of users.

Comment (scruffian) [reply rainfall/style.css]: I removed it

**Review by scruffian — APPROVED** [review]
LGTM

**Pull request merged**